File: src/exam-core/src/main/java/com/wcpdoc/exam/core/entity/MyExam.java
package com.wcpdoc.exam.core.entity;

import java.math.BigDecimal;
import java.util.Date;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;

import lombok.Data;

/**
 * 我的考试实体
 * 
 * v1.0 zhanghc 2017-06-19 16:28:29
 */
@Data
@TableName(value = "EXM_MY_EXAM", autoResultMap = true)
public class MyExam {
	@TableId(type = IdType.AUTO)
	private Integer id;
	private Integer examId;
	private Integer userId;
	private Integer markUserId;
	private Date examStartTime;
	private Date examEndTime;
	private Date markStartTime;
	private Date markEndTime;
	private BigDecimal objectiveScore;
	private BigDecimal totalScore;
	private Integer state;
	private Integer markState;
	private Integer answerState;
	private Integer updateUserId;
	private Date updateTime;
	private Integer no;

	/**
	 * 答题用时（毫秒）<br/>
	 * 分数排名专用
	 * 
	 * v1.0 zhanghc 2023年5月10日上午9:22:57
	 * 
	 * @return Long 毫秒值
	 */
	public Long getAnswerMs() {
		if (this.examStartTime == null || this.examEndTime == null) {
			return Long.MAX_VALUE;
		}
		return this.examEndTime.getTime() - this.examStartTime.getTime();
	}
}

File: h5/src/ts/exam/my-exam.ts
// 我的考试接口
export interface MyExam {
    examId: number | null; // 考试ID
    userId: number | null; // 用户ID
    answerStartTime: string; //答题开始时间（yyyy-MM-dd HH:mm:ss）
    answerEndTime: string; //答题结束时间（yyyy-MM-dd HH:mm:ss）
    markStartTime: string; //阅卷开始时间（yyyy-MM-dd HH:mm:ss）
    markEndTime: string; //阅卷结束时间
    objectiveScore: number | null; //客观题分数
    totalScore: number | null; //总分数
    state: number | null; // 状态（1：未考试；2：考试中；3：已交卷；）
    markState: number | null; //阅卷状态（1：未阅卷；2：阅卷中；3：已阅卷；）
    answerState: number | null; //答题状态（1：及格；2：不及格）
    no: number | null; // 排名
    ver: number | null; // 版本

}
